### PR TITLE
[entityAdapter] handle draft states in selectors

### LINKED ISSUE #606: Idea: wrapping createSelector
Just wanted to get a log of some ideas here in regards to: https://github.com/immerjs/immer/issues/561 and the original issue it references: https://github.com/reduxjs/redux-toolkit/issues/452

@phryneas and I were speaking earlier and brainstormed a few ideas for how this issue could be addressed. Currently, once we bump to v1.4 _(or whichever has immer 7.0.1)_, a user could solve this themselves by doing something like `selectorA(current(state))` inside of a reducer. Being that we use immer everywhere, we could provide a wrapped version of `createSelector` and/or entity adapter selectors that does this automatically for a user. 

Basically, something that would achieve this:
`const wrappedEntitySelector = (value) => isDraft(value) ? entitySelector(current(value)) : entitySelector(value)`

@phryneas I might've forgotten part of that convo, so please correct me if I'm missing something

### PR DESCRIPTION
**Pull request by phryneas**
Closes #606.

@markerikson what are your thoughts on exporting `createDraftSafeSelector`?

### CODE REVIEW

**Comment by markerikson** [comment]
That I could live with.

Y'know, I'm really starting to lose track of all the stuff we _do_ export from RTK... :)

**Comment by phryneas** [comment]
Note to self: export that thing *and* add that to the docs. Tomorrow :)

That TODO list for 1.5.0 is getting shorter and shorter though :)

**Comment by phryneas** [comment]
@markerikson any comments on the docs for this one? :)

File: docs/api/createSelector.mdx [hunk 1]
```diff
@@ -19,3 +19,41 @@ For more details on using `createSelector`, see:
 > **Note**: Prior to v0.7, RTK re-exported `createSelector` from [`selectorator`](https://github.com/planttheidea/selectorator), which
 > allowed using string keypaths as input selectors. This was removed, as it ultimately did not provide enough benefits, and
 > the string keypaths made static typing for selectors difficult.
+
+# `createDraftSafeSelector`
+
+Besides re-exporting `createSelector`, RTK also exports a wrapped version of `createSelector` named `createDraftSafeSelector` that allows you to create selectors that
+can be used in the context of `createReducer` and `createSlice` reducers with mutable logic.
```
Comment (msutkowski) [inline docs/api/createSelector.mdx:26]: Should we explicitly state extra reducers as well?

File: docs/api/createSelector.mdx [hunk 2]
```diff
@@ -19,3 +19,41 @@ For more details on using `createSelector`, see:
 > **Note**: Prior to v0.7, RTK re-exported `createSelector` from [`selectorator`](https://github.com/planttheidea/selectorator), which
 > allowed using string keypaths as input selectors. This was removed, as it ultimately did not provide enough benefits, and
 > the string keypaths made static typing for selectors difficult.
+
+# `createDraftSafeSelector`
+
+Besides re-exporting `createSelector`, RTK also exports a wrapped version of `createSelector` named `createDraftSafeSelector` that allows you to create selectors that
```
Comment (msutkowski) [inline docs/api/createSelector.mdx:25]: Do we need to explain what/why in regards to immer Drafts or link to what isDraftable even is? I think the description you have in the code would actually work well here for clarity.

**Review by msutkowski — COMMENTED** [review]
Looks good! I just have a couple questions on descriptions

**Review by markerikson — CHANGES_REQUESTED** [review]

**Pull request merged**